pkg/signup/service/signup_service_test.go

```diff
@@ -1608,6 +1608,81 @@ func (s *TestSignupServiceSuite) TestGetSignupReadyConditionStatus() {
 	}
 }
 
+func (s *TestSignupServiceSuite) TestGetSignupBannedUserEmail() {
+	// given
+	s.ServiceConfiguration(configuration.Namespace(), true, "", 5)
+
+	us := s.newBannedUserSignup()
+	us.Name = service.EncodeUserIdentifier(us.Spec.IdentityClaims.PreferredUsername)
+	err := s.FakeUserSignupClient.Tracker.Add(us)
+	require.NoError(s.T(), err)
+
+	c, _ := gin.CreateTestContext(httptest.NewRecorder())
+
+	toolchainStatus := s.newToolchainStatus(".apps.")
+	err = s.FakeToolchainStatusClient.Tracker.Add(toolchainStatus)
+	require.NoError(s.T(), err)
+
+	bannedUserID, err := uuid.NewV4()
+	require.NoError(s.T(), err)
+
+	err = s.FakeBannedUserClient.Tracker.Add(&toolchainv1alpha1.BannedUser{
+		TypeMeta: v1.TypeMeta{},
+		ObjectMeta: v1.ObjectMeta{
+			Name:      bannedUserID.String(),
+			Namespace: configuration.Namespace(),
+			Labels: map[string]string{
+				toolchainv1alpha1.BannedUserEmailHashLabelKey: "a7b1b413c1cbddbcd19a51222ef8e20a",
+			},
+		},
+		Spec: toolchainv1alpha1.BannedUserSpec{
+			Email: "jsmith@gmail.com",
+		},
+	})
+	require.NoError(s.T(), err)
+
+	rr := httptest.NewRecorder()
+	ctx, _ := gin.CreateTestContext(rr)
+	ctx.Set(context.UsernameKey, "jsmith")
+	ctx.Set(context.SubKey, us.Spec.IdentityClaims.UserID)
+	ctx.Set(context.EmailKey, "jsmith@gmail.com")
+
+	// when
+	response, err := s.Application.SignupService().GetSignup(c, us.Spec.IdentityClaims.UserID, us.Spec.IdentityClaims.PreferredUsername)
```

Comment:
I'm a little confused (but also, not familiar with the code): `response` is a `*signup.Signup`, so why do we need an HTTP recorder in this test?

Reply:
My understanding is that we need the recorder in order to create the gin context only, so it's just a dependency not actually used. There may be a cleaner way to set this up, but I'm not aware of any atm.